### ISSUE TITLE
compile failure with GHC-9.2.1

### ISSUE THREAD

**Issue by amigalemming** [jkff/storable-endian#5]
~~~~
storable-endian-0.2.6$ PATH=/usr/local/ghc-9.2.1/bin:$PATH cabal build
...
Data/Storable/Endian.hs:370:36: error:
    • Couldn't match expected type ‘Word16#’ with actual type ‘Word#’
    • In the first argument of ‘W16#’, namely
        ‘(w `uncheckedShiftRL#` i)’
      In the expression: W16# (w `uncheckedShiftRL#` i)
      In an equation for ‘shiftr_w16’:
          shiftr_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftRL#` i)
    |
370 | shiftr_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftRL#`   i)
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^

Data/Storable/Endian.hs:371:36: error:
    • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
    • In the first argument of ‘W32#’, namely
        ‘(w `uncheckedShiftRL#` i)’
      In the expression: W32# (w `uncheckedShiftRL#` i)
      In an equation for ‘shiftr_w32’:
          shiftr_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftRL#` i)
    |
371 | shiftr_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftRL#`   i)
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^

Data/Storable/Endian.hs:400:36: error:
    • Couldn't match expected type ‘Word16#’ with actual type ‘Word#’
    • In the first argument of ‘W16#’, namely
        ‘(w `uncheckedShiftL#` i)’
      In the expression: W16# (w `uncheckedShiftL#` i)
      In an equation for ‘shiftl_w16’:
          shiftl_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftL#` i)
    |
400 | shiftl_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftL#`   i)
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^

Data/Storable/Endian.hs:401:36: error:
    • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
    • In the first argument of ‘W32#’, namely
        ‘(w `uncheckedShiftL#` i)’
      In the expression: W32# (w `uncheckedShiftL#` i)
      In an equation for ‘shiftl_w32’:
          shiftl_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftL#` i)
    |
401 | shiftl_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftL#`   i)
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^
~~~~

I wonder whether it is necessary to mess around with the unlifted constructors at all, since `Data.Bits` provides also lifted `uncheckedShift` functions.

**Comment by jkff** [comment]
Oh man, I haven't touched this package in probably a decade. Thank you for the bug report. I'm afraid I don't have time to tackle it myself right now though, but would be happy to approve a PR and re-release the package.

**Comment by jkff** [comment]
Fixed in #6
